rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -43,7 +43,7 @@ trait Reduce[M[_]] {
   def eval(par: Par)(
       implicit env: Env[Par],
       rand: Blake2b512Random,
-      sequenceNumber: Int
+      sequenceNumber: Int = 0
```

Comment:
Default arguments in production code lead to terrible bugs. Please drop this commit.

Reply:
There was an analogous default argument in `ChargingReducer`. Moving it here keeps the refactoring manageable. I can remove it, but a lot of useless zeros will pop up.

Reply:
I see... Those zeros would pop up b/c of duplication in test code. Could you try and introduce test-util methods that don't require the sequence number, and still make the sequenceNumber non-default? Any usage of the default value in prod code we should investigate with high suspicion...

Reply:
I don't know what you mean by duplication in code. Zeros pop up whenever `eval` is called. This is an example:

```
val inspectTaskSendFirst = for {
          _   <- reducer.eval(send)(env, splitRand0)
          _   <- reducer.eval(receive)(env, splitRand1)
          res <- space.toMap
        } yield res
```

usage. There are a total of 208 errors when this default parameter is removed. Given that the default parameter being in `Reduce` leaves the code base in no worse shape than it was when the default parameter was in `ChargingReducer`, I think we should save that fight for another day. I'll leave a TODO.

Reply:
208 errors, of which 200 are in test code - which admittedly shouldn't be passing that parameter and use the default value, contrary to all prod usages. I call 'duplication' the duplicated call to the eval methods, most likely with the same type of assertion, done dozens of times.

A TODO is an improvement :+1: